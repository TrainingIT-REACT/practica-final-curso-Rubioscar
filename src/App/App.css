.App {
  margin: 3em auto;
  max-width: 50em;
}

.App h1 {
  text-align: center;
}

.cardcss.card {
  min-width: 18rem;
}

.navlink {
  color: rgba(255,255,255,.5);
  padding-right: .5rem;
  padding-left: .5rem;
}

.navlink:hover {
  color: rgba(255,255,255,.75);
  text-decoration: none;
}
