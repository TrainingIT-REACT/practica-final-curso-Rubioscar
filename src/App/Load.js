import React, { Component } from 'react';
import Spinner from 'react-bootstrap/Spinner'

class Load extends Component {
  render() {
    return(
      <div>
        <Spinner animation="grow" variant="primary" />
      </div>

    )
  }
}

export default Load;
