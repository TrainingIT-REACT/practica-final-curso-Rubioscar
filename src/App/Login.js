import React, { Component } from 'react';
import Form from 'react-bootstrap/Form';
import Button from 'react-bootstrap/Button';
import { connect } from 'react-redux';

class Login extends Component {
    constructor(props) {
        super(props);

        this.onChangeName = this.onChangeName.bind(this);

        this.state = {
            name: null,
            apellido: null,
            descripcion: null,
        };
    }

    componentDidMount() {
    }

    onSubmit() {
        console.log(this.state.name);
    }

    onChangeName(e) {
        this.setState({ name: e.target.value });
    }

    onChangeApellido(e) {
        console.log(e);
    }

    render() {

      return (
        <div className="App">
            <Form onSubmit={() => this.onSubmit()}>
                <Form.Group>
                    <Form.Label htmlFor="name">
                        Introduce tu nombre
                    </Form.Label>
                    <Form.Control id="name" type="text" value={this.state.name} 
                    placeholder="Nombre" onChange={this.onChangeName}/>
                </Form.Group>
                <Form.Group>
                    <Form.Label>
                        Introduce tu apllido
                    </Form.Label>
                    <Form.Control id="apellido" type="text" value={this.state.name} 
                    placeholder="Apellidos" onChange={(ref) => this.onChangeApellido(ref)}/>
                </Form.Group>
                <Form.Group>
                    <Form.Label>
                        Introduce tu descripcion
                    </Form.Label>
                    <Form.Control id="descripcion" as="textarea" rows="3" value={this.state.name} 
                    placeholder="Descripcion" onChange={this.onChangeName}/>
                </Form.Group>
                <Button type="submit">Logeate!</Button>
            </Form>
        </div>
      );
    }
}

/*const mapStateToProps = (state) => {
  return {
    songs: state.canciones.songs,
    isloading: state.canciones.isLoading,
    loading:  state.albun.isLoading,
    album: state.albun.album,
  }
};*/

/*const mapDispatchToProps = (dispatch) => ({
  fetchSongs: (id) => dispatch(getAlbumSongs(id)),
  fectchAlbum: (id) => dispatch(getAlbum(id)),
});*/

export default (Login);