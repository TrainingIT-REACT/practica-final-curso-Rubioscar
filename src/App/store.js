import { createStore, combineReducers, applyMiddleware } from "redux";
import { createPromise } from 'redux-promise-middleware';
import { createLogger } from 'redux-logger'
import { composeWithDevTools } from 'redux-devtools-extension';

import { albunes, albun } from './reducers/albums';
import { canciones, tendencias, cancion } from './reducers/songs';
import user from './reducers/user';

export default createStore(
    combineReducers({albunes, albun, canciones, tendencias, cancion, user}),
    composeWithDevTools(applyMiddleware(createPromise(), createLogger()))
);