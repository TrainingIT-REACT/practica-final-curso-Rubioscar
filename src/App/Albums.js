import React, { Component } from 'react';
import Card from 'react-bootstrap/Card';
import Button from 'react-bootstrap/Button';
import CardDeck from 'react-bootstrap/CardDeck';

// Css
import './App.css';
import Load from './Load';

class Albums extends Component {
    constructor(props) {
      super(props);
  
      this.state = {
        loading: true,
        albums: []
      }
    }

    async componentDidMount() {
      try {
        const res = await fetch('/albums');
        const json = await res.json();
        this.setState((prevState) => ({
          ...prevState,
          loading: false,
          albums: json
        }));
      } catch(err) {
        console.error("Error accediendo al servidor", err);
      }
    }

    render() {
      return (
        <div className="App">
          <div className="row">
            { this.state.loading ?
                <Load/>
                : <CardDeck>
                    {this.state.albums.map(album => 
                      <Card className="cardcss" style={{ 'margin-bottom': '15px' }}>
                        <Card.Img variant="top" src={album.cover} />
                        <Card.Body>
                          <Card.Title>Autor: {album.artist}</Card.Title>
                          <Card.Text>
                            Cancion: {album.name}
                          </Card.Text>
                          <Button>Entrar</Button>
                        </Card.Body>
                      </Card>
                      )}
                  </CardDeck>
              }
          </div>
        </div>
      );
    }
}

export default Albums;