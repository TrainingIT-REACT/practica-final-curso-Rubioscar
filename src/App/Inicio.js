import React, { Component } from 'react';
import { connect } from 'react-redux';
import Load from './Load';
import Card from 'react-bootstrap/Card';
import Button from 'react-bootstrap/Button';
import CardDeck from 'react-bootstrap/CardDeck';
import { getTrendsSongs } from './actions/songs';
import { setSong } from './actions/historico';

// Css
import './App.css';
import 'bootstrap/dist/css/bootstrap.min.css';

class Inicio extends Component {

  componentDidMount() {
    this.props.fetchTrends();
  }

  onClick(song) {
    this.props.setSongHistory(song);
    this.props.history.push(`/song/${song.id}`);
  }

  render() {
    return (
        <div className="App">
          <h1>Musica Recomendada!</h1>
          <div className="row">
            { this.props.isloading ?
                <Load/>
                : <CardDeck>
                    {this.props.trends.map(song => 
                      <Card className="cardcss" style={{ 'margin-bottom': '15px' }}>
                        <Card.Body>
                          <Card.Title>Name: {song.name}</Card.Title>
                          <Card.Text>
                            Esta canción tiene una duración de:
                                      {song.seconds} sec
                          </Card.Text>
                          <Button onClick={() => this.onClick(song)}>Reproducir</Button>
                        </Card.Body>
                      </Card>
                      )}
                  </CardDeck>
              }
          </div>
        </div>
    );
  }
}

const mapStateToProps = (state) => {
  return {
    trends: state.tendencias.trends,
    isloading: state.tendencias.isLoading,
  }
};

const mapDispatchToProps = (dispatch) => ({
  fetchTrends: () => dispatch(getTrendsSongs()),
  setSongHistory: (song) => dispatch(setSong(song)),
});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(Inicio);
