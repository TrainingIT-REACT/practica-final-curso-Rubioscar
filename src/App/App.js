import React, { Component } from 'react';
//import Load from './Load';
import { BrowserRouter as Router, Route, NavLink } from "react-router-dom";
import Albums from './albunes/Albums';
import Songs from './canciones/Songs';
import Inicio from './Inicio';
import Login from './usuario/Login';
import Perfil from './usuario/Perfil';
import store from './store';
import { Provider } from "react-redux";
import Button from 'react-bootstrap/Button';
import Navbar from 'react-bootstrap/Navbar';
import Nav from 'react-bootstrap/Nav';

// Css
import './App.css';
import 'bootstrap/dist/css/bootstrap.min.css';

class App extends Component {
  /*constructor(props) {
    super(props);

    this.state = {
      loading: true,
      albums: []
    }
  }*/

  /*async componentDidMount() {
    try {
      const res = await fetch('/albums');
      const json = await res.json();
      this.setState((prevState) => ({
        ...prevState,
        loading: false,
        albums: json
      }));
    } catch(err) {
      console.error("Error accediendo al servidor", err);
    }
  }*/

  render() {
    return (
      <Provider store={store}>
        <Router>
          <div className="App">
            <h1>Spotify falso!</h1>
            <Navbar bg="dark" variant="dark">
              <Navbar.Brand href="/home">Falso</Navbar.Brand>
              <Nav className="mr-auto">
                <NavLink to="/home" className="navlink">Home</NavLink>
                <NavLink to="/albums" className="navlink">Albums</NavLink>
                <NavLink to="/user" className="navlink">Perfil</NavLink>
              </Nav>
              <NavLink to="/login">
                <Button variant="outline-info">Login</Button>
              </NavLink>
            </Navbar>
          </div>
          <Route path="/home" exact component={Inicio}/>
          <Route path="/albums" exact component={Albums}/>
          <Route path="/albums/:id" component={Songs}/>
          <Route path="/login" component={Login}/>
          <Route path="/user" component={Perfil}/>
        </Router>
      </Provider>
    );
  }
}

export default App;
