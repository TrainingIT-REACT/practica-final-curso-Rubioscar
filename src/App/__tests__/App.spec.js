import React from 'react';
import { render } from 'enzyme';

// Componente
import App from '../App';
import Load from '../Load';

describe(App, () => {
  it('renders correctly', () => {
    const wrapper = render(<App />);
    expect(wrapper).toMatchSnapshot();
  });
})

describe("Test enzyme de Inicio", () => {
  let wrapper;

  beforeEach(() => {
    wrapper = render(<Load />);
  });

  it('Comprobar la carga de Inicio', () => {
    expect(wrapper.find('div').length).toBe(1);
  })
})
