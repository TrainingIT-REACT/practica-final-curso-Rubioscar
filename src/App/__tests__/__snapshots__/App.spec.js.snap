// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correctly 1`] = `
<div
  class="App"
>
  <h1>
    Spotify falso!
  </h1>
  <nav
    class="navbar navbar-expand navbar-dark bg-dark"
  >
    <a
      class="navbar-brand"
      href="/home"
    >
      Falso
    </a>
    <div
      class="mr-auto navbar-nav"
    >
      <a
        class="navlink"
        href="/home"
      >
        Home
      </a>
      <a
        class="navlink"
        href="/albums"
      >
        Albums
      </a>
      <a
        class="navlink"
        href="/user"
      >
        Perfil
      </a>
    </div>
    <a
      href="/login"
    >
      <button
        class="btn btn-outline-info"
        type="button"
      >
        Login
      </button>
    </a>
  </nav>
</div>
`;
